src/net.h

```diff
@@ -538,6 +539,12 @@ class CConnman
     /** Pointer to this node's banman. May be nullptr - check existence before dereferencing. */
     BanMan* m_banman;
 
+    /**
+     * Addresses that were saved during the previous clean shutdown. We'll
+     * attempt to make block-relay-only connections to them.
```

Comment:
I also wonder if saving anchor don't introduce a slight fingerprint if a node is started on IPV4 then moved to Tor and connect to the same anchors. A spying anchor, by coupling with other infos such as best advertised tip could guess the networks mapping.

I think we should stamp the anchor.dat with last network type connected on and remove then if it has changed compared to actual one.

What's your opinion ?

Reply:
I think it is a right way to only allow connection to an anchor of the network type, which is allowed to connect to by the current policy of the node. Good topic for a follow up.